python/ray/exceptions.py

```diff
@@ -910,6 +907,27 @@ class RayCgraphCapacityExceeded(RaySystemError):
     pass
 
 
+@PublicAPI(stability="alpha")
+class UnpickleableException(RayError):
+    """Raised when attempting to unpickle a serialized exception fails.
+
+    This occurs when deserializing (unpickling) a previously serialized exception
+    fails. In this case, we fall back to raising the string representation of
+    the original exception along with its stack trace that was captured at the
+    time of serialization.
+    """
+
+    def __init__(self, original_stack_trace: str):
+        self.original_stack_trace = original_stack_trace
+
+    def __str__(self):
+        return (
+            "Failed to unpickle serialized exception\n"
```

Comment:
we should include the reason for why the serialized exception couldn't be deserialized here as well. that way it's actionable to fix it

and once the change to enable custom ser/de for exceptions is in, we should add a call to action here